src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -103,6 +105,96 @@ internal sealed partial class LocalRewriter
 
                 return _factory.Sequence(temps.ToImmutableAndFree(), effects.ToImmutableAndFree(), returnValue);
             }
+
+            void reverseAssignmentsToTargetsIfApplicable()
+            {
+                PooledHashSet<Symbol>? visitedSymbols = null;
+
+                Debug.Assert(right is not ({ Kind: BoundKind.TupleLiteral } or BoundConversion { Operand.Kind: BoundKind.TupleLiteral }));
```

Comment:
>BoundKind.TupleLiteral [](http://example.com/codeflow?start=51&length=22)

``` or BoundKind.ConvertedTupleLiteral```?

Reply:
I think we do expect that `right.Kind` can be `ConvertedTupleLiteral` here.